lib/status.js

```diff
@@ -307,16 +309,27 @@
 		};
 
 		checkTimeout = function() {
-			callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
+			return callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
 		};
+		
+		var parsedUrl = URL.parse(req.url),
```

Comment:
you can actually do parsing in one step using:
https://nodejs.org/docs/latest-v0.10.x/api/url.html#url_url_parse_urlstr_parsequerystring_slashesdenotehost
just provide true as second argument and QS params will be parsed as well.